### PR TITLE
Update site to new theme - just-the-docs

### PR DESCRIPTION
**Pull request by maks-oleksyuk**
**Important information!**
The updated theme uses Actions to deploy the site, so you need to set it in the settings.
 - go to the `Settings` tab -> `Pages` -> `Build and deployment`, then select `Source`: `GitHub Actions`.
 - if there were any failed Actions, go to the `Actions` tab and click on `Re-run jobs`.
 
**The main changes made:**
- Images moved to the assets folder
- main information is placed in the footer (license)
- table of contents is now created automatically
- general design changes
- users can now use search
 
 Other information you are interested in can be found in the [documentation](https://just-the-docs.github.io/just-the-docs/)

### CODE REVIEW

File: book/caching.md [hunk 1]
```diff
@@ -1,32 +1,20 @@
...
-  - [Caching REST Resources](#caching-rest-resources)
-  - [Caching in an API class wrapper](#caching-in-an-api-class-wrapper)
-  - [Caching in a .module file](#caching-in-a-module-file)
-  - [Logic for caching render arrays](#logic-for-caching-render-arrays)
-  - [Development Setup](#development-setup)
-    - [Disable caching and enable TWIG debugging](#disable-caching-and-enable-twig-debugging)
-    - [Disable Cache for development](#disable-cache-for-development)
-  - [Reference](#reference)
+## Table of contents
+{: .no_toc .text-delta }
```
Comment (selwynpolit) [inline book/caching.md:12]: Am I understanding correctly that this is a dynamic TOC?

Comment (maks-oleksyuk) [reply book/caching.md:12]: Yes, more about this in [this documentation](https://just-the-docs.github.io/just-the-docs/docs/navigation-structure/#in-page-navigation-with-table-of-contents).

**Comment by selwynpolit** [comment]
@maks-oleksyuk As you can see I merged the MR but I get the following error during build:

1. Branch "gh-pages" is not allowed to deploy to github-pages due to environment protection rules.
2. The deployment was rejected or didn't satisfy other protection rules.

Any thoughts?  I'm looking around at docs to see if I can figure it out.

**Comment by selwynpolit** [comment]
Ah, I think I got it.  I'm going to try change a file and make sure it builds ok.  I think I might prefer a light theme but I will keep it and see if I grow to love it.  Thanks for all the hard work!  Please add yourself to the attributions section.

**Comment by selwynpolit** [comment]
hmmm. Still seeing some errors:

https://github.com/selwynpolit/d9book/actions/runs/4709155962/jobs/8352042683

**Comment by maks-oleksyuk** [comment]
Some settings need to be changed (I wrote about it in the pull request)

go to the `Settings tab` -> `Pages` -> `Build and deployment`, then select Source: GitHub Actions.

This error occurs during the process of publishing a file from GitHub, the current theme uses actions for this

**Comment by selwynpolit** [comment]
Yeah, I did that.  Then changed it back, then selected Github actions again.  I see this:

```
Warning:  github-pages can't satisfy your Gemfile's dependencies.
/usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/theme.rb:84:in `rescue in gemspec': The just-the-docs theme could not be found. (Jekyll::Errors::MissingDependencyException)
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/theme.rb:81:in `gemspec'
	from /usr/local/bundle/gems/jekyll-3.9.3/lib/jekyll/theme.rb:19:in `root'
```

**Comment by maks-oleksyuk** [comment]
I had a similar error.

If the settings are changed to actions, then only action will be run on the next commit (it works successfully).

**Comment by selwynpolit** [comment]
Ah, I did another commit and it got better.  ok, now I can finish up the twig chapter.  Lots of work to do.  Is it difficult to try the light theme?

**Comment by selwynpolit** [comment]
Never mind. I figured it out.  That was easy.

**Comment by selwynpolit** [comment]
@maks-oleksyuk Do you notice that leading backslashes (e.g. as in \Drupal) show up as highlighted.  See a real example at https://selwynpolit.github.io/d9book/reaching-out#guzzle-post-example

Any ideas how to fix that?

**Comment by maks-oleksyuk** [comment]
That's how I saw it. I was thinking of trying to add some custom css files to fix it, or changing the code highlighting to a different theme entirely.
I will try to fix it in the near future

**Pull request merged**